### PR TITLE
Install a Swift 5.8.1 toolchain in all swift-ci images

### PR DESCRIPTION
**Pull request by bnbarham**
Installs a Swift 5.8.1 toolchain into `/opt/swift/<version>` on each Linux platform. Removes clang since the toolchain clang can be used instead.

### CODE REVIEW

**Review by DougGregor — APPROVED** [review]
There's no way to share this code across Dockerfiles?

**Comment by bnbarham** [comment]
@shahmishal has some docker-compose ideas here IIUC.

With the base docker... not really. You could share a common base image, but this is the other way around - ie. we have a bunch of images that we then want to add a toolchain on top of. Could do some generator-like hackery I suppose, but I'm not sure that's really better (eg. could just have a makefile that concats a bunch of files together 😅). Another alternative would be something like edrevo/dockerfile-plus.

**Comment by shahmishal** [comment]
@swift-ci test

File: swift-ci/master/amazon-linux/2/Dockerfile [hunk 1]
```diff
@@ -47,6 +46,51 @@ RUN mkdir -p /usr/local/lib/python3.7/site-packages/
...
+    && gpg --batch --quiet --keyserver keyserver.ubuntu.com --recv-keys "$SWIFT_SIGNING_KEY" \
+    && gpg --batch --verify swift.tar.gz.sig swift.tar.gz \
+    # - Unpack the toolchain, set libs permissions, and clean up.
+    && mkdir -p $SWIFT_PREFIX \
+    && tar -xzf swift.tar.gz --directory $SWIFT_PREFIX --strip-components=1 \
+    && chmod -R o+r $SWIFT_PREFIX/usr/lib/swift \
+    && rm -rf "$GNUPGHOME" swift.tar.gz.sig swift.tar.gz
+
+ENV PATH="${SWIFT_PREFIX}/usr/bin:${PATH}"
+
```
Comment (tomerd) [inline swift-ci/master/amazon-linux/2/Dockerfile:91]: should we use the RPM packages instead?

**Comment by tomerd** [comment]
> There's no way to share this code across Dockerfiles?

some of this can be made into a script that is that can be copied & used in the docker file

**Comment by bnbarham** [comment]
> > There's no way to share this code across Dockerfiles?
> 
> some of this can be made into a script that is that can be copied & used in the docker file

Ah that's a good point. Could do that, the extra curl for Ubuntu is a little annoying but we could install based on a parameter for that script.

**Comment by tomerd** [comment]
> the extra curl for Ubuntu is a little annoying but we could install based on a parameter for that script.

a bit less efficient from docker caching pov but we can install curl on ubuntu before the ADD and RUN commands for the script, then remove after. or even a more lazy approach, we can just include curl on these images in the main dependencies list even if we dont really need it for anything else

File: swift-ci/master/amazon-linux/2/Dockerfile [hunk 2]
```diff
@@ -2,7 +2,7 @@ FROM amazonlinux:2
 
 RUN yum install shadow-utils -y
 
-RUN groupadd -g 998 build-user && \
+RUN groupadd -g 1000 build-user && \
```
Comment (MaxDesiatov) [inline swift-ci/master/amazon-linux/2/Dockerfile]: I don't have any opinion on the group number, just curious why this change was made?

Comment (bnbarham) [reply swift-ci/master/amazon-linux/2/Dockerfile]: I cannot remember why this was needed. It definitely was for CentOS when using the release image, but I don't know if it is still needed now that we're not. I'm guessing probably not? I don't suppose you remember @shahmishal / @DougGregor?

Comment (DougGregor) [reply swift-ci/master/amazon-linux/2/Dockerfile]: I have no memory of this place

Comment (bnbarham) [reply swift-ci/master/amazon-linux/2/Dockerfile]: I've reverted for now. We'll see if there's any errors pretty quick.

**Comment by MaxDesiatov** [comment]
@swift-ci test

File: swift-ci/master/amazon-linux/2/Dockerfile (hunk 1 above)
Comment (MaxDesiatov) [reply swift-ci/master/amazon-linux/2/Dockerfile:91]: We currently mark those as "experimental use only" on swift.org/download. Are we willing to try those out here anyway, or is description on that page outdated?

**Comment by bnbarham** [comment]
@swift-ci test

**Comment by bnbarham** [comment]
@swift-ci please test

**Comment by bnbarham** [comment]
@swift-ci please test
